src/frontend/src/components/common/Tooltip/index.tsx

```diff
@@ -0,0 +1,36 @@
+import { useState, useRef } from 'react';
+import { TooltipProps } from '../Tooltip/interfaces';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+
+/**
+ * Tooltip component that provides additional information on hover.
+ * @param {TooltipProps} props - Props for the Tooltip component.
+ */
+export const Tooltip = (props: TooltipProps) => {
+  const { children, content, direction, id } = props;
+  const [visible, setVisible] = useState(false);
+  const ref = useRef(null);
+
+  return (
+    <div
+      className={'tooltip-container'}
+      onMouseEnter={() => setVisible(true)}
+      onMouseLeave={() => setVisible(false)}
+      ref={ref}
+    >
+      <div className={`tooltip-content-wrapper tooltip-${direction}`}>
```

Comment:
Maybe we need a prop for showIcon because not all tooltips will require the icon, like when we implement status tooltip